### PR TITLE
Add 1.24.0 to CI integration tests

### PR DESCRIPTION
**Pull request by tomhjp**
Drop 1.20 and add the latest 1.24 image for integration testing.

### CODE REVIEW

**Review by swenson — APPROVED** [review]
LGTM

**Review by tvoran — COMMENTED** [review]
The test failure is troubling, but `kind create cluster --image kindest/node:v1.24.0` doesn't work for me locally either.

**Comment by tomhjp** [comment]
Looks like `kind` v0.13.0 fixes the issues with running v1.24.0: https://github.com/kubernetes-sigs/kind/releases/tag/v0.13.0

File: integrationtest/integration_test.go [hunk 1]
```diff
@@ -32,28 +34,36 @@ func TestMain(m *testing.M) {
...
+	}
+}
+
+func getTokenReviewerJWT() string {
+	return runCmd("kubectl --namespace=test create token test-token-reviewer-account")
 }
 
 func getVaultServiceAccountJWT() string {
-	return runCmd("kubectl exec --namespace=test vault-0 -- cat /var/run/secrets/kubernetes.io/serviceaccount/token")
+	return runCmd("kubectl --namespace=test create token vault")
```
Comment (tvoran) [inline integrationtest/integration_test.go:63]: Ooh, this is cool

**Review by tvoran — APPROVED** [review]

**Pull request merged**